src/main/java/de/uhd/ifi/se/decision/management/jira/quality/MetricCalculator.java

```diff
@@ -0,0 +1,402 @@
+	}
+	return numberOfSentencesPerIssue;
+    }
+
+    public Map<String, Integer> getDistributionOfKnowledgeTypes() {
+	Map<String, Integer> distributionOfKnowledgeTypes = new HashMap<String, Integer>();
+	for (KnowledgeType type : KnowledgeType.getDefaultTypes()) {
+	    int numberOfElements = persistenceManager.getDecisionKnowledgeElements(type).size();
+	    for (DecisionKnowledgeElement element : (Optional.ofNullable(decisionKnowledgeCodeElements)
+		    .orElse(Collections.emptyList()))) {
+		if (element.getType().equals(type)) {
+		    numberOfElements++;
+		}
+	    }
+	    for (DecisionKnowledgeElement element : (Optional.ofNullable(decisionKnowledgeCommitElements)
+		    .orElse(Collections.emptyList()))) {
+		if (element.getType().equals(type)) {
+		    numberOfElements++;
+		}
+	    }
+	    for (Issue issue : jiraIssues) {
+		if (issue.getIssueType().getName().equals(type.toString())) {
+		    numberOfElements++;
+		}
+	    }
+	    distributionOfKnowledgeTypes.put(type.toString(), numberOfElements);
+	}
+	return distributionOfKnowledgeTypes;
+    }
+
+    public Map<String, Integer> getReqAndClassSummary() {
+	Map<String, Integer> summaryMap = new HashMap<String, Integer>();
+	int numberOfRequirements = 0;
+	for (Issue issue : jiraIssues) {
+	    // Temporary Solution until Settings are available
+	    if (issue.getIssueType().getName().equals("System Function")
+		    || issue.getIssueType().getName().equals("Nonfunctional Requirement")
+		    || issue.getIssueType().getName().equals("Persona")
+		    || issue.getIssueType().getName().equals("Usertask")
+		    || issue.getIssueType().getName().equals("Subtask")
+		    || issue.getIssueType().getName().equals("Workspace")) { // Temporary Solution until Settings are
+									     // available
+		numberOfRequirements++;
+	    }
+	}
+	summaryMap.put("Requirements", numberOfRequirements);
+	GitCodeClassExtractor extract = new GitCodeClassExtractor("CONDEC");
+	summaryMap.put("Code Classes", extract.getNumberOfCodeClasses());
+	return summaryMap;
+    }
+
+    public Map<String, Integer> getKnowledgeSourceCount() {
+	Map<String, Integer> sourceMap = new HashMap<String, Integer>();
+	if (decisionKnowledgeCodeElements != null) {
+	    sourceMap.put("Code", decisionKnowledgeCodeElements.size());
+	} else {
+	    sourceMap.put("Code", 0);
+	}
+	if (decisionKnowledgeCommitElements != null) {
+	    sourceMap.put("Commit", decisionKnowledgeCommitElements.size());
+	} else {
+	    sourceMap.put("Commit", 0);
+	}
+	sourceMap.put("Issue Content", persistenceManager.getDecisionKnowledgeElements().size());
+	int numberOfElements = 0;
+	for (Issue issue : jiraIssues) {
+	    if (KnowledgeType.getDefaultTypes().toString().contains(issue.getIssueType().getName())) {
+		numberOfElements++;
+	    }
+	}
+	sourceMap.put("Jira Issues", numberOfElements);
+	return sourceMap;
+    }
+
+    public Map<String, String> getDecKnowlElementsOfATypeGroupedByHavingElementsOfOtherType(KnowledgeType linkFrom,

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getDecKnowlElementsOfATypeGroupedByHavingElementsOfOtherType(KnowledgeType, KnowledgeType)' has an NPath complexity of 440](https://app.codacy.com/app/UHD/cures-condec-jira/pullRequest?prid=4666670)